File: models/Pago.cs
namespace TallerMecanico.models;

public class Pago
{
    public int Id { get; set; }
    public int ClienteId { get; set; }
    public Cliente Cliente { get; set; }
    public decimal Monto { get; set; }
    public DateTime FechaTransaccion { get; set; }

    // Relación con el objeto pagado (Producto, Servicio, Parte)
    public TipoPago TipoPago { get; set; }
    public int? ProductoId { get; set; }   // Ahora es opcional
    public Producto Producto { get; set; }
    public int? ServicioId { get; set; }   // Ahora es opcional
    public Servicio Servicio { get; set; }
    public int? ParteId { get; set; }      // Ahora es opcional
    public Parte Parte { get; set; }

    // Auditoría
    public DateTime FechaCreacion { get; set; }
    public DateTime? FechaActualizacion { get; set; }
    public DateTime? FechaBorrado { get; set; }
    public bool EstaBorrado { get; set; } = false;
}


public enum TipoPago
{
    Producto,
    Servicio,
    Parte
}

File: AFront/src/app/Entity/Pago.ts
export interface Pago {
  id: number;
  clienteId: number;
  monto: number;
  fechaTransaccion: Date;
  tipoPago: TipoPago; // enum que puedes definir en TypeScript
  productoId: number | null;
  servicioId: number | null;
  parteId: number | null;
  fechaCreacion: Date;
  fechaActualizacion: Date | null;
  fechaBorrado: Date | null;
  estaBorrado: boolean;
}

// Enum para tipo de pago
export enum TipoPago {
  Producto = 0,
  Servicio = 1,
  Parte = 2
}
